ts/config.ts

```diff
@@ -93,4 +93,13 @@ var config = AWS.config.loadFromPath('/to/path');
 // test update allowing unknown keys
 AWS.config.update({
    fake: 'fake' 
-}, true);
\ No newline at end of file
+}, true);
+
+// Test constructing with a CredentialProviderChain
+var options = {
+    credentialProvider: new AWS.CredentialProviderChain([
+        () => new AWS.EC2MetadataCredentials()
+    ])
+};
+var s3 = new AWS.S3(options);
+var ses = new AWS.SES(options);
```

Comment:
@RLovelett 
This should be fine. We really just want to make sure the code compiles correctly, and that we don't break it in the future. This captures the intent.